lib/bloc_record/selection.rb

```diff
@@ -1,17 +1,348 @@
+		SQL
+
+		rows_to_array(rows)
+	end
+
+
+	def method_missing(method, *arguments, &block)
+		if method == "find_by_name"
+			find_by(:name, *arguments)
+		else
+			puts "no such method found"
+		end
+	end
+
+	def find_each(attribute, value, batch_size=nil, &block)
+		# set batch of records to array rows
+		# if batch_size not nil and is an integer and not 0 or less, set Limit on record 
+		if attribute && attribute.is_a?(String) && value
+	
+			if batch_size && batch_size.is_a?(Integer)
+				if batch_size < 1
+					puts "Invalid batch size"
+				else
+					rows = connection.execute <<-SQL
+						SELECT #{columns.join ","} FROM #{table}
+						WHERE #{attribute} = #{BlocRecord::Utility.sql_strings(value)}
+						LIMIT #{batch_size};
+					SQL
+				end
+			elsif batch_size
+				puts "Invalid data type for batch size parameter. Try again with a positive integer."
+			else				
+				rows = connection.execute <<-SQL
+					SELECT #{columns.join ","} FROM #{table}
+					WHERE #{attribute} = #{BlocRecord::Utility.sql_strings(value)};
+				SQL
+			
+				objArray = rows_to_array(rows)
+				transformedArray = []
+
+				for obj in objArray
+					transformedArray << block.call(obj)
+				end
+
+				transformedArray
+			end
+		else
+			puts "Attribute must be a string and value cannot be null for find_each"
+		end
+	end
+
+
+	def find_in_batches(start, batch_size, &block)
+		if start && batch_size && start.is_a?(Integer) && batch_size.is_a?(Integer)
+			if start >= 0 && batch_size > 0
+				rows = connection.execute <<-SQL
+					SELECT #{columns.join ","} FROM #{table}
+					LIMIT #{batch_size} 
+					OFFSET #{start};
+				SQL
+
+				objArray = rows_to_array(rows)
+
+				yield(objArray, batch_size)
+			else
+				puts "start must be at least zero and batch size greater than zero"
+			end
+		else
+			puts "start, batch_size arguments can't be nil and must be integers."
+		end
+	end
+
+	def where(*args)
+		# Array - .count here filters for arrays vs other data types
+		#  .shift pops the first element off the array, shrinking it, and returns it
+		# String accepts ("phone_number = '555-222-1010'") input
+		#  with the string case, params is nil. We put the full string in the expression variable
+		# For Hash, we convert Symbols in keys to Strings
+		#  then we convert values to SQL-friendly strings and connect multiple clauses with AND
+		if args.count > 1
+			expression = args.shift
+			params = args
+		else
+			case args.first
+			when String
+				expression = args.first
+			when Hash
+				expression_hash = BlocRecord::Utility.convert_keys(args.first)
+				expression = expression_hash.map {|key, value| "#{key}=#{BlocRecord::Utility.sql_strings(value)}"}.join(" AND ")
+			end
+		end
+
+		sql = <<-SQL
+			SELECT #{columns.join ","} FROM #{table}
+			WHERE #{expression};
+		SQL
+
+		rows = connection.execute(sql, params)
+		rows_to_array(rows)
+	end
+
+	def order(*args)
+		# .count filters for arrays, to_s handles with String, Symbol cases
```

Comment:
You miss the `Hash` case:

`Entry.order(name: :asc, phone_number: :desc)`

Believe it or not, that will be passed in as a `Hash`.